packages/fakta-medlemskap/src/v3/components/situasjon/situasjonUtils.ts

```diff
@@ -0,0 +1,71 @@
+import { MedlemskapV3, UtlandsoppholdPeriode } from '@navikt/fp-types';
+import { getKodeverknavnFn, KodeverkType, opplysningAdresseType } from '@navikt/fp-kodeverk';
+import { alleKodeverk } from '@navikt/fp-storybook-utils';
+import kodeverkTyper from '@navikt/fp-kodeverk/src/kodeverkTyper';
+import { IntlShape } from 'react-intl';
+import { TIDENES_ENDE } from '@navikt/ft-utils';
+
+export function sorterPerioder<T extends { fom: string; tom: string }>(a: T, b: T) {
+  const dateAEnd = a.tom ? new Date(a.tom) : new Date(TIDENES_ENDE);
+  const dateBEnd = b.tom ? new Date(b.tom) : new Date(TIDENES_ENDE);
+
+  if (dateAEnd > dateBEnd) return -1;
+  if (dateAEnd < dateBEnd) return 1;
+
+  const dateAStart = new Date(a.fom);
+  const dateBStart = new Date(b.fom);
+
+  if (dateAStart > dateBStart) return -1;
+  if (dateAStart < dateBStart) return 1;
+
+  return 0;
+}
+
+const toTitleCapitalization = (sentence: string): string => {
+  return sentence
+    .toLowerCase()
+    .split(' ')
+    .map(word => word.charAt(0).toUpperCase() + word.slice(1)) // Capitalize the first letter of each word
+    .join(' ');
+};
+
+function formaterLand(landNavn: string, intl: IntlShape) {
+  return intl.formatMessage({ id: 'Situasjon.ILand' }, { land: toTitleCapitalization(landNavn) });
+}
+
+export function getSisteRegion(medlemskap: MedlemskapV3): string {
+  const alleRegioner = alleKodeverk[KodeverkType.REGION];
+  const nyesteRegion = medlemskap.regioner.sort(sorterPerioder)[0];
+  return alleRegioner.find(r => r.kode === nyesteRegion.type)?.navn ?? '-';
+}
+
+export function getSistePersonstatus(medlemskap: MedlemskapV3): string {
+  const nyeste = medlemskap.personstatuser.sort(sorterPerioder)[0];
+  if (nyeste) {
+    const personstatus = getKodeverknavnFn(alleKodeverk)(nyeste.type, KodeverkType.PERSONSTATUS_TYPE);
+    if (personstatus) return personstatus;
+  }
+  return '-';
+}
+
+export function getSisteBostedsLand(medlemskap: MedlemskapV3, intl: IntlShape): string {
+  const nyeste = medlemskap.adresser
+    .filter(adresse => adresse.adresse.adresseType === opplysningAdresseType.BOSTEDSADRESSE)
+    .sort(sorterPerioder)[0]?.adresse.land;
+  if (nyeste) {
+    const landNavn = getKodeverknavnFn(alleKodeverk)(nyeste, kodeverkTyper.LANDKODER);
+    if (landNavn) return formaterLand(landNavn, intl);
+  }
+  return '-';
+}
+
+export function formaterUtenlandsopphold(utenlandsopphold: UtlandsoppholdPeriode[], intl: IntlShape): string {
+  if (utenlandsopphold.length == 1) {
+    return formaterLand(utenlandsopphold[0].landNavn, intl);
+  } else if (utenlandsopphold.length > 1) {
+    return intl.formatMessage({ id: 'Situasjon.FlereLand' });
+  } else {
```

Comment:
var ein eslint-rule som kalla dette for unødvendig else. Vart sikkert fjerna da eg oppdaterte til eslint 9. Men er ikkje så nøye for min del korleis du skriv dette altso

Reply:
jeg fikset det!